src/Raven.Server/Documents/Handlers/Batches/Commands/MergedBatchCommand.cs

```diff
@@ -465,12 +469,20 @@ protected override long ExecuteCmd(DocumentsOperationContext context)
                     tpl.Reply[tpl.FieldName] = changeVector;
             }
         }
+
+        // We are requested to do not return result.
+        if (IncludeReply == false)
+        {
+            Reply.Clear();
```

Comment:
`Debug.Assert(IncludeReply == false && Reply.Count == 0)`

Reply:
The point here was that this list should be empty, so we want to remove that Clear.

Each place that executes:
```
Reply.Add(sth);
```

Should be replaced with:
```
if (IncludeReply)
    Reply.Add(sth);
```

Reply:
Ahh I didnt understood. I thought you wanted to add an extra check for the invariant under that case.